### ISSUE TITLE
change the cost function

### ISSUE THREAD

**Issue by freddycct** [dmlc/mxnet#2472]
Hi, 

How do I know what is the cost function used in the output layers? Can I change it to least squares error for LinearRegressionOutput?

**Comment by tornadomeet** [comment]
```python
model.fit(
        X                  = train,
        eval_data          = val,
        eval_metric        = ['rmse'],
        ...)
```

**Comment by freddycct** [comment]
this is the loss for evaluation but not for the training, i thought

**Comment by tornadomeet** [comment]
@freddycct No, train and test also can use it, just try

**Comment by pinglforever** [comment]
When you choose mx.symbol.LinearRegressionOutput as last layer, the train loss function is encoded in that layer. `eval_metric = ['rmse']` is just for the train and val result feedback for us to tune the hyperparameters

**Comment by freddycct** [comment]
ok thank you
